llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -3370,6 +3391,20 @@ void CallsiteContextGraph<DerivedCCG, FuncTy, CallTy>::identifyClones(
 
   assert(Node->AllocTypes != (uint8_t)AllocationType::None);
 
+  DenseSet<uint32_t> RecursiveContextIds;
+  // If we are not skipping recursive callsites, and have also enabled
+  // skipping of recursive contexts, look for context ids that show up in
+  // multiple caller edges.
+  if (!SkipRecursiveCallsites && SkipRecursiveContexts) {
+    DenseSet<uint32_t> AllCallerContextIds;
+    for (auto &CE : Node->CallerEdges) {
+      AllCallerContextIds.reserve(CE->getContextIds().size());
```

Comment:
`AllCallerContextIds` is reused across each CallerEdge without being cleared. I guess the intent here is to reserve it to the size of the largest one? Can you document the intent here?

Reply:
yep, done